### DISCUSSION TITLE
Error when using integrate_output()

### DISCUSSION THREAD

**Discussion by MattDamiano-NOAA** [vast-lib/tinyVAST#55] [General]
Hi Jim, 

My student and I have been working on adapting the bivariate spatiotemporal model demo code to fit to black sea bass and red snapper count data from mounted video component of the Southeast Reef Fish Survey. Per our previous offline conversation, we ran the following code with a reduced mesh size:

dat <- read.csv("video.dataset.2011.2022.subset_tiny.csv")

Data <- data.frame(time = dat$Time, x = dat$Start_Longitude, y= dat$Start_Latitude, 
                   var = dat$Species, n = dat$Counts)

n_t = 12

# make mesh
mesh = fm_mesh_2d( Data[,c('x','y')], cutoff=0.1, refine=TRUE )

# Define DSEM
dsem = "
  rs -> rs, 1, b11
  bsb -> bsb, 1, b22
  bsb -> rs, 1, b21
  rs -> bsb, 1, b12
  rs <-> rs, 0, var1
  bsb <-> bsb, 0, var1
"
# fit model
out = tinyVAST( dsem = dsem,
                data = Data,
                formula = n ~ 0 + var,
                spatial_graph = mesh,
                family = tweedie(),
                control = tinyVASTcontrol( trace=1 )
              )

After the model successfully converges, we do get results for parameter and coefficient estimates. However, when we tried applying the following code to produce the estimated indices of relative abundance,

Est1 = sapply( seq_len(n_t), FUN=\(t) integrate_output(out, newdata=subset(Data,time==t & var=="rs")) )
Est2 = sapply( seq_len(n_t), FUN=\(t) integrate_output(out, newdata=subset(Data,time==t & var=="bsb")) )

we received the following error: 

Error in cbind(ans, rbind(NA * ans1, NA * ans2, ans4)) : 
  number of rows of matrices must match (see arg 2)

Any thoughts about what is generating these NAs, and/or general guidance on using the integrate_output function? I think that the model is set up correctly in that our data only contain the nominal counts (n) vs. the simulated mean (z) and mean plus obs error (n) in the simulation example code.

Cheers, 

Matt

**Comment by James-Thorson-NOAA** [comment]
It appears that your predictive data `Data` does not have years indexed `1:n_t`.  Please try:

```R
sapply( unique(Data$t), FUN=\(t) integrate_output(out, newdata=subset(Data,time==t & var=="rs")) )
```

closing for now, but please tell me if that doesn't fix it.
